src/pokedex_area_screen.c

```diff
@@ -1,20 +1,1437 @@
+            InitRegionMap(&gPokedexAreaScreenPtr->regionMap, FALSE);
+            StringFill(gPokedexAreaScreenPtr->charBuffer, CHAR_SPACE, 16);
+            break;
+        case 2:
+            ResetDrawAreaGlowState();
+            break;
+        case 3:
+            if (DrawAreaGlow())
+                return;
+            break;
+        case 4:
+            CreateRegionMapPlayerIcon(1, 1);
+            RegionMapDefaultZoomOffsetPlayerSprite(0, -8);
+            SetVBlankCallback(VBlankCB_AreaScren);
+            break;
+        case 5:
+            BeginNormalPaletteFade(0xFFFFFFEB, 0, 16, 0, 0);
+            break;
+        case 6:
+            REG_BLDCNT = BLDCNT_TGT1_BG0 | BLDCNT_EFFECT_BLEND | BLDCNT_TGT2_BG0 | BLDCNT_TGT2_BG1 | BLDCNT_TGT2_BG2 | BLDCNT_TGT2_BG3 | BLDCNT_TGT2_OBJ | BLDCNT_TGT2_BD;
+            StartAreaGlow();
+            REG_DISPCNT = DISPCNT_MODE_1 | DISPCNT_OBJ_1D_MAP | DISPCNT_BG0_ON | DISPCNT_BG1_ON | DISPCNT_BG2_ON | DISPCNT_OBJ_ON | DISPCNT_WIN0_ON | DISPCNT_WIN1_ON;
+            break;
+        case 7:
+            SetCallback(DebugCB_WaitFade);
+            SetMainCallback2(MainCB_AreaScren);
+            return;
+    }
+    gMain.state++;
+}
+
+static void VBlankCB_AreaScren(void)
+{
+    LoadOam();
+    ProcessSpriteCopyRequests();
+    TransferPlttBuffer();
+}
+
+static void MainCB_AreaScren(void)
+{
+    gPokedexAreaScreenPtr->callback();
+    DoAreaGlow();
+    AnimateSprites();
+    BuildOamBuffer();
+    UpdatePaletteFade();
+}
+
+static void SetCallback(void (*func)(void))
+{
+    gPokedexAreaScreenPtr->callback = func;
+    gPokedexAreaScreenPtr->state = 0;
+}
+
+static void ResetDrawAreaGlowState(void)
+{
+    gPokedexAreaScreenPtr->drawAreaGlowState = 0;
+}
+
+bool8 DrawAreaGlow(void)
+{
+    switch (gPokedexAreaScreenPtr->drawAreaGlowState)
+    {
+        case 0:
+            FindMapsWithMon(gPokedexAreaScreenPtr->species);
+            break;
+        case 1:
+            BuildAreaGlowTilemap();
+            break;
+        case 2:
+            LZ77UnCompVram(gUnknown_083F8438, BG_CHAR_ADDR(3));
+            break;
+        case 3:
+            DmaCopy16(3, gPokedexAreaScreenPtr->areaGlowTilemap, BG_SCREEN_ADDR(30), 0x500);
+            break;
+        case 4:
+            LoadPalette(gUnknown_083F8418, 0, 32);
+            break;
+        case 5:
+            REG_BG0CNT = BGCNT_PRIORITY(1) | BGCNT_CHARBASE(3) | BGCNT_16COLOR | BGCNT_SCREENBASE(30) | BGCNT_TXT256x256;
+            gPokedexAreaScreenPtr->drawAreaGlowState++;
+            return FALSE;
+        default:
+            return FALSE;
+    }
+    gPokedexAreaScreenPtr->drawAreaGlowState++;
+    return TRUE;
+}
+
+static void FindMapsWithMon(u16 mon)
+{
+    u16 i;
+    struct Roamer *roamer;
+
+    if (mon != ROAMER_SPECIES)
+    {
+        gPokedexAreaScreenPtr->numOverworldAreas = 0;
+        gPokedexAreaScreenPtr->numSpecialAreas = 0;
+        for (i = 0; i < 1; i++)
+        {
+            if (sWynautData[i] == mon)
```

Comment:
`sWynautData` looks more like `sSpeciesHiddenFromDexMaps`.  Also, probably want to use `ARRAY_LENGTH` for the for loop's bounds.